src/qml/MessagesListView.qml

```diff
@@ -90,8 +90,8 @@ MultipleSelectionListView {
 
         Component.onCompleted: {
             var sourceFile = ""
-            var isMmsError = model.textMessageType === HistoryEventModel.MessageTypeMultiPart && model.textMessageAttachments.length === 0 && model.textMessage.length === 0
-            if (isMmsError) {
+            var isIncomingMmsError = model.senderId !== "self" && model.textMessageType === HistoryThreadModel.MessageTypeMultiPart && model.textMessageAttachments.length === 0 && (model.textMessage.length === 0 || model.textMessageStatus === HistoryThreadModel.MessageStatusTemporarilyFailed || model.textMessageStatus === HistoryThreadModel.MessageStatusPermanentlyFailed || model.textMessageStatus === HistoryThreadModel.MessageStatusPending)
```

Comment:
@jezek so we don't need anymore the empty mesage check if i follow well ? (model.textMessage.length === 0)

Reply:
I would leave it there. So it will work on possible empty messages, which were meant as error.
If I used only the status check, all previous error messages, introduced by you, were not recognized as error messages.
The empty message, should be handled as permanent error state, which can't be redownloaded.